packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/useLocationDetailView.ts

```diff
@@ -119,16 +139,25 @@ export function useLocationDetailView(
   React.useEffect(() => {
     if (hasInvalidPrefix) return;
     handleList({
+      config: getConfig(),
       prefix: key,
       options: { ...listOptions, refresh: true },
     });
     handleReset();
-  }, [handleList, handleReset, listOptions, hasInvalidPrefix, prefix, key]);
+  }, [
+    handleList,
+    handleReset,
+    listOptions,
```

Comment:
`listOptions` is a ref so it shouldn't need to be in the deps array and `prefix` is not used in the effect afaict?

Reply:
`listOptions` changing would result in a rerun of the effect which is not the desired the behavior for an initial load?

Reply:
I agree, but we're never changing the ref in this hook. And eslint complains if i exclude `listOptions`

Reply:
Not needed to merge but that is due to assigning `listOptionsRef.current` to a variable. The dep can be removed by direct usage of the ref in the effect:
```
...listOptionsRef.current,
// other options
```